src/routes/tools/extreme-heat/_data.js

```diff
@@ -42,113 +197,76 @@ export async function get98pThreshold(climvar, queryParams) {
   }
 }
 
-export async function addModel(config, params, modelId) {
-  const { climvarId, scenarioId } = config;
-  const slugs = [
-    `${climvarId}_day_${modelId}_historical`,
-    `${climvarId}_day_${modelId}_${scenarioId}`,
-  ];
-  const promises = slugs.map((slug) => {
-    return pipe(fetchTimeseries, transformResponse)({ slug, params });
-  });
-  return Promise.all(promises)
-    .then((results) => {
-      const values = merge(results);
-      if (values.length === 0) {
-        throw new Error("No Data");
-      }
-      const series = {
-        key: modelId,
-        type: "line",
-        values,
-      };
-      return addSeriesInfo(series);
-    })
-    .catch((error) => {
-      throw new Error(`${modelId}: ${error.message}`);
-    });
-}
+/**
+ * The following 4 functions transform the data for different extreme heat indicators
+ * @param {array} series
+ * @return {array}
+ */
 
-export async function getObserved(config, params) {
-  try {
-    const { climvarId } = config;
-    const slug = `${climvarId}_day_livneh`;
-    const promise = pipe(fetchTimeseries, transformResponse)({ slug, params });
-    const values = await promise;
-    if (values.length === 0) {
-      throw new Error("No Data");
-    }
-    const observed = {
-      key: "observed",
-      type: "line",
-      values: values.filter((d) => d.date.getFullYear() < 2007),
-    };
-    return addSeriesInfo(observed);
-  } catch (error) {
-    throw new Error(`Observed: ${error.message}`);
+// Group daily data by year
+// Backfill data for missing years with empty array
+export const groupDataByYear = (series) => {
```

Comment:
I'm using the `events` endpoint for the Extreme heat tool which returns the raw data (in this case - temperature) for all dates. With the `thresh` param it will return data values only above the threshold. I calculate all the indicators locally, e.g. count/max duration, etc. So it behaves a little differently than the `exheat`/`cdd`/`hdd` endpoints. I think in this case it should not be backfilled with 0 from the API.

Reply:
For the instances of `new Date(year, 0, 1)` it might be good to do `new Date(Date.UTC(year, 0, 1))` instead to avoid timezone issues.